src/controllers/connectionGroupWebviewController.ts

```diff
@@ -120,3 +128,72 @@ export function createConnectionGroupFromSpec(spec: ConnectionGroupState): IConn
         id: Utils.generateGuid(),
     };
 }
+
+/**
+ * Opens the connection group dialog with the initial state.
+ */
+export function openConnectionGroupDialog(state) {
+    state.dialog = {
+        type: "createConnectionGroup",
+        props: {},
+    } as CreateConnectionGroupDialogProps;
+
+    return state;
+}
+
+/**
+ * Shared function for controllers to create a connection group from the provided spec.
+ * This function will add the group to the connection store and update the form state of the controller.
+ * @param connectionGroupSpec - The specification for the connection group to create.
+ * @param connectionStore - The connection store to add the group to.
+ * @param telemetryView - The telemetry view to send events to.
+ * @param state - The form state of the controller.
+ * @param formErrorObject - An object to store any error messages that occur during the creation process.
+ * @param connectionProfile - The connection profile to update with the new group ID.
+ * @return A promise that resolves to the updated form state.
+ */
+export async function createConnectionGroup(
+    connectionGroupSpec: ConnectionGroupSpec,
+    connectionStore: ConnectionStore,
+    telemetryView: TelemetryViews,
+    state: FormState<any, any, any>,

```

Comment:
This'll change with the other comments but for future ref:

Instead of `<any, any, any>`, `function createConnectionGroup<TState, TFormTarget, etc.>()`